sql/core/benchmarks/CollationBenchmark-jdk21-results.txt

```diff
@@ -1,54 +1,63 @@
-UNICODE_CI                                       318228         318726         705          0.0     3182275.2       0.0X
+UTF8_BINARY_LCASE                                 10164          10165           2          0.0      101635.6       1.0X
+UNICODE                                            5683           5684           1          0.0       56828.5       1.8X
+UTF8_BINARY                                        6280           6281           2          0.0       62802.3       1.6X
+UNICODE_CI                                       307901         317477       13542          0.0     3079007.4       0.0X
 
-OpenJDK 64-Bit Server VM 21.0.3+9-LTS on Linux 6.5.0-1018-azure
+OpenJDK 64-Bit Server VM 21.0.3+9-LTS on Linux 6.5.0-1021-azure
 AMD EPYC 7763 64-Core Processor
 collation unit benchmarks - endsWith:     Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-UTF8_BINARY_LCASE                                 11655          11664          12          0.0      116552.8       1.0X
-UNICODE                                            6235           6239           5          0.0       62350.8       1.9X
-UTF8_BINARY                                        7066           7069           5          0.0       70658.1       1.6X
-UNICODE_CI                                       313515         313999         685          0.0     3135149.1       0.0X
+UTF8_BINARY_LCASE                                 10360          10361           1          0.0      103596.7       1.0X
+UNICODE                                            5667           5668           0          0.0       56674.0       1.8X
+UTF8_BINARY                                        6307           6309           3          0.0       63069.2       1.6X
+UNICODE_CI                                       311942         312293         496          0.0     3119419.4       0.0X
+
+OpenJDK 64-Bit Server VM 21.0.3+9-LTS on Linux 6.5.0-1021-azure
+AMD EPYC 7763 64-Core Processor
+collation unit benchmarks - mode - 30105 elements:  Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
+---------------------------------------------------------------------------------------------------------------------------------
+UTF8_BINARY_LCASE - mode - 30105 elements                      4              4           0         80.4          12.4       1.0X
+UNICODE - mode - 30105 elements                                0              0           0       1277.7           0.8      15.9X
+UTF8_BINARY - mode - 30105 elements                            0              0           0       1282.2           0.8      15.9X
+UNICODE_CI - mode - 30105 elements                             9              9           0         32.5          30.7       0.4X
```

Comment:
Replace with:
https://github.com/GideonPotok/spark/actions/runs/9120613791

```

OpenJDK 64-Bit Server VM 21.0.3+9-LTS on Linux 6.5.0-1021-azure
AMD EPYC 7763 64-Core Processor
collation unit benchmarks - mode - 30105 elements:  Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
---------------------------------------------------------------------------------------------------------------------------------
UTF8_BINARY_LCASE - mode - 30105 elements                     40             41           0          7.5         133.6       1.0X
UNICODE - mode - 30105 elements                                3              3           0        115.2           8.7      15.4X
UTF8_BINARY - mode - 30105 elements                            3              3           0        114.4           8.7      15.3X
UNICODE_CI - mode - 30105 elements                           191            193           1          1.6         635.9       0.2X
```

Reply:
The best performance we have seen is with a OpenHashMap approach, basically forcing binary equality. But there are a lot of pitfalls. So we should get this in place and think about whether that is necessary.

We can consider switching with that implementation later.